pkg/prompts/prompts.go

```diff
@@ -153,9 +171,19 @@ func RunDefaultableStringPrompt(customPrompt config.BuilderVar, defaultValue str
 	if err != nil {
 		return "", err
 	}
-	// Variable-level substitution, we need to get defaults so later references can be resolved in this loop
-	if input == "" && defaultString != "" {
-		input = defaultValue
+	if customPrompt.Name == "APPNAME" {
```

Comment:
i don't see a reason to have custom logic for app name. wouldn't a validate make more sense to make them choose a proper name instead of sanitizing user input?

Reply:
yeah, I have this question too! 
I'm sanitizing the App name for any non-allowed characters. I will check with David and if its irrelevant to sanitize user input I will update the logic.

Reply:
we already take a `validate` func to this method, so we should pass a validator for the APPNAME variable instead of having the logic inside the code that runs for every prompt